tests/Fixer/Whitespace/BlankLineBeforeStatementFixerTest.php

```diff
@@ -397,6 +397,50 @@ public function provideFixWithDefaultCases()
         ];
     }
 
+    /**
+     * @dataProvider provideFixWithDocCommentCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithDocCommentCases($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'statements' => ['doc_comment'],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideFixWithDocCommentCases()
+    {
+        return [
+            [
+                '<?php
+/** @var int $foo */
```

Comment:
Why there is no line added before this doc comment?

Reply:
Because that is the file's doc-comment, which normally contains annotations like `@author`, it belongs at the top.
The purpose of this fixer is to separate doc-comments from other declarations above it, like here:
```php
$foo = 123;
/** @var int $baz */
$baz = 456;
```
The top-doccomment does not have another declaration above it, just the file's beginning (which it is normally describing). What should the file-doccomment be separated from?

I will revert the changes to `logo.png` as soon as I have a bit of time to set everything up again, most likely in the next few days. Maybe today. Thank you for the review.

Reply:
> What should the file-doccomment be separated from?

Could be from openning tab, but AFAIR there is other fixer to do that, also the same applies to code, not only PHPDoc.